### PR TITLE
Add a setting for SDK download

### LINKED ISSUE #1433: Add a setting for SDK generator
Add a setting for SDK generator in platform settings (Settings view, settings collection).

# User story
> As an Apinf administrator I want to be able to add a Codegen instance location to Apinf
so that I can allow API consumers to download SDK's 
whenever APIs added to Apinf have a Swagger file.

# Definition of done
* [ ] Settings collection contains a setting for SDK generator.
* [ ] Settings view contains a setting for SDK generator.
* [ ] An Apinf administrator can give the location of the SDK generator (= Codegen).
* [ ] The SDK generator UI in Documentation tab passes the Swagger to the Codegen instance defined in Settings.
* [ ]  SDK generator is available in Documentation tab only if there is a setting for SDK generator

### PR DESCRIPTION
**Pull request by marla-singer**
Closes #1433
- Add fileds for setting Codegen instance in Setting platform: collection and view
- Change condition for button 'SDK Generator': Button is available if there is a setting for SDK generator
- Change host from ```generator.swagger.io``` to the Codegen instance defined in Settings.

### CODE REVIEW

**Comment by brylie** [comment]
Please be aware that @jykae is restructuring the Documentation module folder to follow our recently added [folder structure guidelines](https://github.com/apinf/api-umbrella-dashboard/blob/develop/CONTRIBUTING.md#file-structure).

The relevant part is as follows:

- **/component_name**
  - **/client**
    - **/sub_component**
      - *sub_component.html*
      - *sub_component.js*
    - **/sub_component_2**
      - *sub_component_2.html*
      - *sub_component_2.js*

This prevents sub-components from needing their own `client` folders, since they are under a shared client folder.

**Comment by marla-singer** [comment]
Ok, I will pull changes by @jykae for Documentation module

File: documentation/client/view/documentation.js [hunk 1]
```diff
@@ -37,36 +38,58 @@ Template.documentation.helpers({
       const currentDocumentationFileId = currentDocumentationFile._id;
 
       // Get documentation file URL
-      return Meteor.absoluteUrl().slice(0, -1) + DocumentationFiles.baseURL + "/id/" + currentDocumentationFileId;
+      return Meteor.absoluteUrl().slice(0, -1) + DocumentationFiles.baseURL + '/id/' + currentDocumentationFileId;
     }
   },
-  documentationLink: function() {
+  documentationLink: function () {
```
Comment (brylie) [inline documentation/client/view/documentation.js]: Use ES2015 enhanced object literal notation here:

```js
documentationLink () {
  ...
},
```

File: documentation/client/view/documentation.js [hunk 2]
```diff
@@ -37,36 +38,58 @@ Template.documentation.helpers({
...
     // get documentation link
     const documentationLink = this.apiBackend.documentation_link;
     // check if exists
     if (documentationLink) {
-      return documentationLink
+      return documentationLink;
     }
   },
   documentationExists: function () {
```
Comment (brylie) [inline documentation/client/view/documentation.js]: Use [ES2015 enhanced object literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#Object_literals) notation here:

```js
documentationExists () {
  ...
},
```

File: documentation/client/view/documentation.js [hunk 3]
```diff
@@ -37,36 +38,58 @@ Template.documentation.helpers({
...
     const currentApiBackend = this.apiBackend;
     if (currentApiBackend.documentationFileId) {
       return true;
     }
+  },
+  codegenServerExists: function () {
```
Comment (brylie) [inline documentation/client/view/documentation.js]: Use [ES2015 enhanced object literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#Object_literals) notation here:

```js
codegenServerExists () {
  ...
},
```

**Comment by brylie** [comment]
This PR looks good. There are a couple of cleanup and consistency improvements, noted above.

**Comment by brylie** [comment]
Also, consider linting this PR with ESLint. We have an open PR for ESLint configuration #1466, which you can use as a basis.

**Comment by brylie** [comment]
@marla-singer please resolve conflicts. We merged a refactoring branch that touched some of the documentation module files.

**Comment by brylie** [comment]
Thanks. We are trying to organize our code for consistency lately, so things are moving around as we are working.

**Comment by brylie** [comment]
When I try to save the Codegen setting, the form does not appear to submit. I click the button, but nothing happens. No error is visible in the console.

**Comment by brylie** [comment]
There appears to be a regression in the development branch.

**Comment by marla-singer** [comment]
When I tested this feature on my old localhost version, it worked correct. 
But now I did ``` meteor reset``` and had this problem too. However another fields on this page don't work too.

**Comment by brylie** [comment]
It appears related to switching the Settings collection file to using export. I have a fix in progress.

**Comment by brylie** [comment]
See PR #1475

**Comment by brylie** [comment]
The fix may not be as simple as I originally thought. I will have to continue working on this tomorrow.

**Pull request merged**